packages/shared/src/utils/convertToMarkdownLinks.ts

```diff
@@ -0,0 +1,6 @@
+// Regular expression to match URLs
+const urlRegex = /(?:https|ftp):\/\/[^\s/$.?#].[^\s]*[^\s.,!?#$]/g;
```

Comment:
how come we wanna check for / allow ftp specifically?

Reply:
maybe we can use the `VALID_URL` regex pattern defined in util/regex.ts in web (can move here), we use that one right now to detect urls when they get pasted into the collection description/bio editor I believe

that one is a bit more flexible by allowing urls that start with `www` instead of `https://` but also bit more strict about disallowing special characters in the domain

Reply:
@kaitoo1 updated the regex and test to allow for links starting with "www" and disallow "ftp" but did not use the `VALID_URL` because I think @Robinnnnn mentioned on slack thread that we should allow p much any link and that logic actually disallows some of the links in the test which seem like common use cases.